pycbc/waveform/waveform.py

```diff
@@ -775,6 +775,73 @@ def get_td_waveform_from_fd(rwrap=0.2, **params):
                                            params['f_lower']))
     return hp, hc
 
+def get_td_det_waveform_from_fd_det(template=None, rwrap=0.2, **params):
+    """ Return time domain version of fourier domain approximant which
+    includes detector response, with padding and tapering at the start
+    of the waveform.
+
+    Parameters
+    ----------
+    rwrap: float
+        Cyclic time shift parameter in seconds. A fudge factor to ensure
+        that the entire time series is contiguous in the array and not
+        wrapped around the end.
+    params: dict
+        The parameters defining the waveform to generator.
+        See `get_fd_det_waveform_sequence`.
+
+    Returns
+    -------
+    dict
+        The detector-frame waveform (with detector response) in time
+        domain. Keys are requested data channels.
+    """
+    kwds = props(template, **params)
+    # determine the duration to use
+    full_duration = duration = pnutils.get_imr_duration(
+            m1=kwds['mass1'], m2=kwds['mass2'], s1z=kwds['spin1z'],
+            s2z=kwds['spin1z'], f_low=kwds['f_lower'],
+            approximant="IMRPhenomD")
+    nparams = kwds.copy()
+
+    while full_duration < duration * 1.5:
+        full_duration = duration = pnutils.get_imr_duration(
+                m1=kwds['mass1'], m2=kwds['mass2'], s1z=kwds['spin1z'],
+                s2z=kwds['spin1z'], f_low=kwds['f_lower'],
+                approximant="IMRPhenomD")
+        nparams['f_lower'] *= 0.99
+
+    if 'f_ref' not in nparams:
+        nparams['f_ref'] = params['f_lower']
+
+    # factor to ensure the vectors are all large enough. We don't need to
+    # completely trust our duration estimator in this case, at a small
+    # increase in computational cost
+    fudge_duration = (max(0, full_duration) + .1 + rwrap) * 1.5
+    fsamples = int(fudge_duration / kwds['delta_t'])
+    N = pnutils.nearest_larger_binary_number(fsamples)
+    fudge_duration = N * kwds['delta_t']
+
+    nparams['delta_f'] = 1.0 / fudge_duration
+    wfs = get_fd_det_waveform_sequence(**nparams)
```

Comment:
Does it seem right to call a sequency version here? Shouldn't we be calling something like get_fd_det_waveform? If we don't have that, shouldn't we? Should the plugin advertise support for that? (and maybe we need to add that to the list of interfaces if we don't already have it). 

What do you think? Wouldn't that be more consistant?